src/com/simulator/controller/SmartHomeSimulatorController.java

```diff
@@ -300,24 +303,100 @@ private void setHouseView() {
+        this.allRoomsListViewZone.getItems().remove(selectedItem);
+    }
 
+    @FXML
+    private void removeRoomFromSelectedRoomsPanel(){
+        Object selectedItem = this.selectedRoomsListForZone.getSelectionModel().getSelectedItem();
+        this.selectedRoomsListForZone.getItems().remove((String) selectedItem);
+        this.allRoomsListViewZone.getItems().add(selectedItem);
     }
     @FXML
-    private void removeRoomFromZoneFunction(){
+    private void addPresetTemperatureToZone(){
+        if(temperatureTextField.getText() ==null || temperatureTextField.getText().trim().isEmpty()||!temperatureTextField.getText().matches("[0-9]+")){
+            Logger.getInstance().outputToConsole("Please make sure a valid temperature is entered");
+            return;
+        }
+        int tempPreset = Integer.valueOf(temperatureTextField.getText());
+        if(listZones.getSelectionModel().isEmpty()){
+            Logger.getInstance().outputToConsole("Please make sure you have selected a zone");
+            return;
+        }
+        String selectedZone = (String)listZones.getSelectionModel().getSelectedItem();
+        if(temperatureComboBox.getSelectionModel().isEmpty()){
+            Logger.getInstance().outputToConsole("Please make sure you have selected a period of time");
+            return;
+        }
+        String selectedPeriodOfTheDay = (String)temperatureComboBox.getSelectionModel().getSelectedItem();
 
+        //Set the new temperature, unless one has already been overwritten for a specific room
+        heatingModule.setTempForZone(selectedZone, selectedPeriodOfTheDay, tempPreset);
+        System.out.println("Came out the other side brother");
     }
+
     @FXML
-    private void setRoomsZonePanel(){
-        ArrayList<String> roomNameList = (ArrayList<String>) House.getInstance().getRoomsNameList();
+    private void createZoneButton(){
+        //Check to see if the user selected some entries
+        if(selectedRoomsListForZone.getItems().size()==0){
+            Logger.getInstance().outputToConsole("Please select some entries for the zone");
+            return;
+        }
 
-        for (String room: roomNameList){
-            allRoomsListViewZone.getItems().add(room);
+        //check to see if the string already exists or if an invalid length string was entered
+        if(zoneName.getText().length()<1||!heatingModule.checkIfValidZoneName(zoneName.getText())){
+            Logger.getInstance().outputToConsole("Please enter a valid name");
+            return;
         }
-    }
 
+        //Create a new ArrayList containing each of the selected rooms
+        String newZoneName = zoneName.getText();
+        ObservableList<String> newZoneObservableList = (ObservableList<String>) this.selectedRoomsListForZone.getItems();
+        ArrayList<Room> newRoomArrayList = new ArrayList<Room>();
+        for(int i=0; i<newZoneObservableList.size(); ++i){
```

Comment:
Same here
```suggestion
        for(int i = 0; i < newZoneObservableList.size(); ++i){
```

and is there a reason you are incrementing ++i instead of i++ does it do the same thing in a for loop?

Reply:
yes it does the same thing in for loops